sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild

```diff
@@ -0,0 +1,39 @@
+# Copyright 2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit linux-mod
+
+GIT_COMMIT="c8313abffe083ac63bf76d2cc90d3edf5b2d1188"
+
+DESCRIPTION="PCIE RTS5209 card reader driver for Linux"
+HOMEPAGE="https://github.com/gexplorer/RTS5209-linux-driver"
+SRC_URI="https://github.com/gexplorer/RTS5209-linux-driver/archive/${GIT_COMMIT}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+DEPEND=""
+RDEPEND="${DEPEND}"
+
+S="${WORKDIR}/RTS5209-linux-driver-${GIT_COMMIT}"
+
+MODULE_NAMES="rts_pstor(misc/drivers/scsi)"
+BUILD_TARGETS="default"
+CONFIG_CHECK="~!MISC_RTSX_PCI"
+ERROR_MISC_RTSX_PCI="CONFIG_MISC_RTSX_PCI: The in-kernel driver rtsx_pci is configured, which may have the same functionality than this driver. Make sure that your kernel loads only one of both drivers - use a module blacklisting."
+
+src_prepare() {
+	default
+
+	sed -e 's#make -C \/lib\/modules\/\$(shell uname -r)\/build\/#+make -C \$(KERNELDIR)#' -i Makefile || die "sed failed!"
```

Comment:
This really shouts for a patch. Also, do you really need to pass it explicitly? I dare say other modules work without such hacks.

Reply:
I mean other kernel modules than this one. They get this right without having to hack it over in every  ebuild.

Reply:
I tried it out - patch file is ~700 byte in size. sed instruction + BUILD_PARAMS setting together ~200 byte. You choose, please.

Reply:
You can create smaller patches by controlling context. Sed instruction has the main problem that it returns successfully even if it doesn't substitute anything.

Reply:
converted to minimum-style patch